README.md

```diff
@@ -7,5 +7,68 @@
 [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 [![Imports: isort](https://img.shields.io/badge/%20imports-isort-%231674b1?style=flat&labelColor=ef8336)](https://pycqa.github.io/isort/)
 
+## What and why
 
-Application_config is a module for configuring a python application.
+Application_config is a module for configuring a python application. It uses toml 
+configuration files that are parsed into dataclasses.
+This brings some benefits:
+
+- Configuration parameters are typed
+- IDEs will provide helpful hints and completion when using configuration parameters
+- More control over what happens when a config file contains incorrect or malicious text
+- Possibility to specify defaults when no config file is found
+
+## How
+
+### Define config section(s) and the container with application info
+
+Example:
+
+```python
+from application_config import (
+    ConfigBase,
+    ConfigSectionBase,
+)
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class ExampleConfigSection(ConfigSectionBase):
+    """Config section for an example"""
+
+    field1: str = "field1"
+    field2: int = 2
+
+
+@dataclass(frozen=True)
+class ExampleConfig(ConfigBase):
+    """Config for an example"""
+
+    section1: ExampleConfigSection = ExampleConfigSection()
+
+    @staticmethod
+    def get_app_basename() -> str:
+        """Return the string that describes the application base name"""
+        return "example"
```

Comment:
I think we should should explicitly mention that the `get_app_basename` determines the folder in the HOME directory the config file should be placed.

Reply:
Maybe that could even go in the docstring? But it should at least explicitly be mentioned in the readme i think.

Reply:
Done